modules/packages/Search.chpl

```diff
@@ -159,7 +157,7 @@ proc linearSearch(Data:[?Dom], val, comparator:?rec=defaultComparator, lo=Dom.lo
    :rtype: (`bool`, `Dom.idxType`)
 
  */
-proc binarySearch(Data:[?Dom], val, comparator:?rec=defaultComparator, in lo=Dom.low, in hi=Dom.high) {
+proc binarySearch(Data:[?Dom], val, comparator:?rec=defaultComparator, in lo=Dom.low, in hi=Dom.alignedHigh) {
```

Comment:
I'm guessing that `lo` should be `Dom.alignedLow` as well for symmetry arguments (e.g., if we had 1..100 by 10 align 9, I could imagine that there could be a similar infinite loop searching at the bottom of the range).

And probably ditto for both args of linear search as well (e.g., if someone passed in a domain that was `[1..10 by 2 aligned 2]` would we search 1, 3, 5, 7, 9 rather than 2, 4, 6, 8, 10?

Reply:
Good catch!